### DISCUSSION TITLE
How to install mToolkit for Eclipse IDE Version: 2024-03 (4.31.0)

### DISCUSSION THREAD

**Discussion by alexcorvis84** [eclipse-kura/kura#5232] [Q&A] [answered]
Hi all,

I've following the [latest docs](https://eclipse.github.io/kura/docs-release-5.4/java-application-development/hello-world-application/#export-the-osgi-bundle) in order to configure my setup with Eclipse IDE Version: 2024-03 (4.31.0). 

I'm stuck at **Create a Deployment Package** step, as it seems that **mtoolkit** needs to be installed but there's no update reference in the docs about how to deal with this or the steps that need to be taken.

Can anyone kindly please tell me how can I install mtoolkit in Eclipse IDE in order to let me Create a Deployment Package?

Many thanks in advance.

**Comment by alexcorvis84** [comment]
Surfing on the web, I've checked in [ESF docs](https://esf.eurotech.com/docs/development-environment-setup#section-installing-mtoolkit) that there's a reference:

`the tool is unmaintained by the author and we suggest to use maven plugin to generate the deployment packages.`

So could you please share a step by step manual how to Create a Deployment Package within that plugin? Also, it would be nice to **update** the **[Hello World Application](https://eclipse.github.io/kura/docs-release-5.4/java-application-development/hello-world-application/#overview)** example in docs accordingly and up to date please.

**Comment by pierantoniomerlino** [comment]
Hi @alexcorvis84, thank you for pointing out the issue on the documentation. We'll update it asap.

Regarding the dp generation, take a look [here](https://github.com/eclipse/kura/blob/1ea36ff8d103186f63be27fa46dd738508699f3e/kura/examples/pom.xml#L437). Kura uses the [osgi-dp plugin ](https://ctron.github.io/osgi-dp/) to generate the deployment package during the build.

**Reply by alexcorvis84** [reply]
Thank you for replying @pierantoniomerlino 

Despite the links shared, the process is not clear to me as I don't have much knowledge about maven and its plugins configuration. So I hope you can update the documentation according to the process to follow. 

Please let me know when it is ready. Thanks in advance!

**Reply by alexcorvis84** [reply, marked as answer]
Just a quick update: I've been able to finally generate the dp with the osgi-dp plugin using Maven 🥳  Thanks!
